editoast/src/core/conflict_detection.rs

```diff
@@ -78,16 +78,65 @@ pub struct ConflictDetectionResponse {
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
 pub struct PacedTrainOccurrenceId {
     paced_train_id: i64,
-    index: i64,
+    index: i32,
 }
 
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, ToSchema)]
 pub struct Conflict {
+    /// List of train schedule ids and paced train generated occurrences involved in the conflict
+    pub train_ids: Vec<String>,
+    /// List of work schedule ids involved in the conflict
+    pub work_schedule_ids: Vec<String>,
+    /// Datetime of the start of the conflict
+    pub start_time: DateTime<Utc>,
+    /// Datetime of the end of the conflict
+    pub end_time: DateTime<Utc>,
+    /// Type of the conflict
+    #[schema(inline)]
+    pub conflict_type: ConflictType,
+    /// List of requirements causing the conflict
+    pub requirements: Vec<ConflictRequirement>,
+}
+
+impl Conflict {
+    pub fn adapt_conflict(self, key: HashMap<String, (i64, Option<i32>)>) -> ConflictResponse {
```

Comment:
What do you think about replacing `(i64, Option<i32>)` (which can be a bit confusing) with : 
```rust
enum TrainId {
    TrainSchedule(int),
    PacedTrainOccurrence(PacedTrainOccurrenceId), // Could be also: PacedTrainOccurence{ id: i64, index: u64 }
}
```